parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -292,14 +292,14 @@ write_files:
   content: |
     [Unit]
     Description=Installs and loads Nvidia GPU kernel module
+    Before=kubelet.service
     [Service]
     Type=oneshot
     RemainAfterExit=true
     ExecStartPre=/bin/sh -c "dkms autoinstall --verbose"
     ExecStart=/bin/sh -c "nvidia-modprobe -u -c0"
-    ExecStartPost=/bin/sh -c "sleep 10 && systemctl restart kubelet"
     [Install]
-    WantedBy=multi-user.target
+    WantedBy=multi-user.target kubelet.service
```

Comment:
we want to move away from saving things through custom data, I think you will need to update it somewhere else

Reply:
parts/linux/cloud-init/artifacts/nvidia-modprobe.service

^ I think this should be source of truth. And this file will override the content on VM during provision time.

Reply:
Ah nevermind, it's already done before. I should read the PR description first.

Reply:
I tend to agree it's chaotic having two sources for this service. I'd like to remove the custom data but following the Chesterton's fence principle I don't like to remove things without first knowing why they are there in the first place. Do you think it's safe to remove the nvidia-modprobe service from custom data? I'm not sure if there is a reason it's duplicated in the first place.